tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -117,10 +129,43 @@ def _check_cache_topology(test_microvm, num_vcpus_on_lvl_1_cache,
                             expected_level_3_topology)
 
 
-@pytest.mark.skipif(
-    platform.machine() != "x86_64",
-    reason="Firecracker supports topology and feature masking only on x86_64."
-)
+def _check_cache_topology_arm(no_cpus=1):
+    # We will check the cache topology by looking at what each cpu
+    # contains as far as cache info.
+    # For that we are iterating through the hierarchy of folders inside:
+    # /sys/devices/system/cpu/cpuX/cache/indexY/type - the type of the cache
+    # (i.e Instruction, Data, Unified)
+    # /sys/devices/system/cpu/cpuX/cache/indexY/size - size of the cache
+    # /sys/devices/system/cpu/cpuX/cache/indexY/level - L1, L2 or L3 cache.
+    # There are 2 types of L1 cache (instruction and data) that is why the
+    # "cache_info" variable below has 4 items.
+
+    path = "/sys/devices/system/cpu/"
+    cache_info = [{"level": "1", "size": "64K", "type": "Data"}, {"level": "1", "size": "64K", "type": "Instruction"},
```

Comment:
As I mentioned in the previous comments, I think this should assert against host topology.